src/osd/OSDMap.cc

```diff
@@ -2599,6 +2619,19 @@ int OSDMap::object_locator_to_pg(
   return map_to_pg(loc.get_pool(), oid.name, loc.key, loc.nspace, &pg);
 }
 
+tl::expected<pg_t, int> OSDMap::object_locator_to_expected_pg(
+  const object_t& oid,
+  const object_locator_t& loc) const
```

Comment:
Your new map_to_pg variant is simply an overload with the same name -- I'd expect this to be parallel.  I wouldn't mind changing both object_locator_to_pg overloads to object_locator_to_raw_pg, though.